apps/vs-agent/src/utils/WebVhDidRegistrar.ts

```diff
@@ -0,0 +1,284 @@
+export class WebVhDidRegistrar implements DidRegistrar {
+  supportedMethods: string[] = ['webvh']
+
+  /**
+   * Updates an existing DID document and its log in the repository.
+   * Uses internal logic to validate verification methods and handle errors.
+   * @param agentContext The agent context.
+   * @param options The update options, including DID, log, signer, verifier, and services.
+   * @returns The result of the DID update, with error handling and validation.
+   */
+  public async update(agentContext: AgentContext, options: WebVhDidUpdateOptions): Promise<DidUpdateResult> {
+    return this._update(agentContext, { ...options })
+  }
+
+  /**
+   * Internal update logic for a DID document.
+   * Handles validation, crypto instance setup, and repository update.
+   * @param agentContext The agent context.
+   * @param options The update options, including signer and verifier.
+   * @returns The result of the DID update, with error handling.
+   */
+  private async _update(
+    agentContext: AgentContext,
+    options: WebVhDidUpdateOptions & { signer?: Signer; verifier?: Verifier },
+  ): Promise<DidUpdateResult> {
+    try {
+      const { did, domain, didDocument: inputDidDocument } = options
+      const didRepository = agentContext.dependencyManager.resolve(DidRepository)
+      const [didRecord] = await didRepository.getCreatedDids(agentContext, { did, method: 'webvh' })
+      const log = didRecord.metadata.get('log') as any[]
+      if (!log) {
+        return {
+          didDocumentMetadata: {},
+          didRegistrationMetadata: {},
+          didState: {
+            state: 'failed',
+            reason: 'The log registry must be created before it can be edited',
+          },
+        }
+      }
+
+      const {
+        controller,
+        authentication,
+        assertionMethod,
+        keyAgreement,
+        service,
+        verificationMethod: inputVerificationMethod,
+      } = inputDidDocument
+      const verificationMethods =
+        inputVerificationMethod ?? (log[log.length - 1].state.verificationMethod as VerificationMethod[])
+      const updateKeys = log[log.length - 1].parameters.updateKeys
+      if (!verificationMethods || verificationMethods.length === 0) {
+        return {
+          didDocumentMetadata: {},
+          didRegistrationMetadata: {},
+          didState: {
+            state: 'failed',
+            reason: 'At least one verification method must be provided.',
+          },
+        }
+      }
+
+      // Always use parseCryptoInstance to get signer/verifier
+      const { signer, verifier } = await this.parseCryptoInstance(
+        agentContext,
+        verificationMethods[0].publicKeyMultibase!,
+        {
+          signer: options.signer,
+          verifier: options.verifier,
+        },
+      )
+
+      if (!signer || !verifier) {
+        throw new Error(
+          `No crypto instances available for DID: ${did}. Provide signer and verifier in options.`,
+        )
+      }
```

Comment:
I don't think this whole code block and parseCryptoInstance make sense now that we are not using any pool of crypto instances. It could be simply replaced by:

```suggestion
    const signer = options.signer ?? new WebvhDidCryptoSigner(agentContext, verificationMethods[0].publicKeyMultibase)
    const verifier = options.verifier ?? new WebvhDidCrypto(agentContext)
```

And do some check of existance of verificationMethods[0].publicKeyMultibase, to avoid using the !